neard/src/runtime.rs

```diff
@@ -350,9 +354,9 @@ impl NightshadeRuntime {
             receipt_result,
             validator_proposals: apply_result.validator_proposals,
             total_gas_burnt,
-            total_validator_reward: apply_result.stats.total_validator_reward,
-            total_balance_burnt: apply_result.stats.total_balance_burnt
-                + apply_result.stats.total_balance_slashed,
+            total_balance_burnt: apply_result.stats.tx_burnt_amount
```

Comment:
Can this overflow? For this the runtime should receiver an invalid state with validator slashed for full u128 amount. Unlikely to build it, but maybe possible eventually.

Reply:
Hm, how is this possible? Validators won't have full u128 amount and we only slash what they have, so this slashed amount is by definition only up to the sum of total stake which is < total_supply * (1 + inflation)^(over time).

I'll wrap it anyway, but curious that I'm not missing anything in the logic here.

Reply:
It might be part of the challenge or some other way to pass an invalid state transition. In the current Runtime implementation we treat any value as it could possibly contain any other value.

Reply:
well, it seems like before it wasn't treated like this as it wasn't wrapped 🤷